BurstBotNET/Commands/ChaseThePig/Game.cs

```diff
@@ -0,0 +1,899 @@
+            })
+            .ToImmutableArray();
+
+        var descriptions = newExposures
+            .Select(e => e switch
+            {
+                ChasePigExposure.Transformer => localization.ExposeTransformer.Replace("{ratio}", 2.ToString()),
+                ChasePigExposure.DoubleGoat => localization.ExposeGoat.Replace("{ratio}", 2.ToString()),
+                ChasePigExposure.DoubleMinus => localization.ExposeHeartA.Replace("{ratio}", 2.ToString()),
+                ChasePigExposure.DoublePig => localization.ExposePig.Replace("{ratio}", 2.ToString()),
+                ChasePigExposure.FirstTransformer => localization.ExposeTransformer.Replace("{ratio}", 4.ToString()),
+                ChasePigExposure.FirstDoubleGoat => localization.ExposeGoat.Replace("{ratio}", 4.ToString()),
+                ChasePigExposure.FirstDoubleMinus => localization.ExposeHeartA.Replace("{ratio}", 4.ToString()),
+                ChasePigExposure.FirstDoublePig => localization.ExposePig.Replace("{ratio}", 4.ToString()),
+                _ => ""
+            })
+            .ToImmutableArray();
+        
+        await using var renderedCard = SkiaService.RenderDeck(deckService, exposedCards);
+
+        foreach (var (_, player) in oldGameState.Players)
+        {
+            if (player.TextChannel == null) continue;
+
+            var randomFileName = Utilities.GenerateRandomString() + ".jpg";
+            var randomAttachmentUri = $"attachment://{randomFileName}";
+
+            var title = localization.ExposeTitle
+                .Replace("{playerName}", player.PlayerId == previousPlayerNewState.PlayerId
+                    ? localizations.GetLocalization().GenericWords.Pronoun
+                    : previousPlayerNewState.PlayerName);
+
+            var formattedTitles = titles
+                .Select(s => s.Replace("{playerName}", player.PlayerId == previousPlayerNewState.PlayerId
+                    ? localizations.GetLocalization().GenericWords.Pronoun
+                    : previousPlayerNewState.PlayerName));
+
+            var description = string.Join('\n', formattedTitles) + "\n\n" + string.Join('\n', descriptions);
+
+            var embed = new Embed(
+                Author: new EmbedAuthor(previousPlayerNewState.PlayerName, IconUrl: previousPlayerNewState.AvatarUrl),
+                Title: title,
+                Description: description,
+                Colour: BurstColor.Burst.ToColor(),
+                Thumbnail: new EmbedThumbnail(Constants.BurstLogo),
+                Image: new EmbedImage(randomAttachmentUri));
+
+            await using var imageCopy = new MemoryStream((int)renderedCard.Length);
+            await renderedCard.CopyToAsync(imageCopy);
+            renderedCard.Seek(0, SeekOrigin.Begin);
+            imageCopy.Seek(0, SeekOrigin.Begin);
+
+            var attachments = new OneOf<FileData, IPartialAttachment>[]
+            {
+                new FileData(randomFileName, imageCopy)
+            };
```

Comment:
I feel like this thing can be simplified into another method or some sort

Reply:
A helper maybe

Reply:
So I gave it a try and it's....a little bit complicated. `using` statement is a syntactic sugar to ensure `IDisposable`'s `Dispose()` is correctly called even if an exception happens, while `await using` will asynchronously call `IAsyncDisposable`'s `DisposeAsync()`. From the official reference (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement), an `using` statement will be translated to:
```csharp
string manyLines = @"This is line one
This is line two
Here is line three
The penultimate line is line four
This is the final, fifth line.";

{
    var reader = new StringReader(manyLines);
    try
    {
        string? item;
        do
        {
            item = reader.ReadLine();
            Console.WriteLine(item);
        } while (item != null);
    }
    finally
    {
        reader?.Dispose();
    }
}
```
That means the lifetime of whatever it is in the `using` statement will be limited to that method call. Making it to another method means:

1. You cannot use `using` statement in that helper method as the stream will be disposed after it returns to the caller. And in the case of attachments, even if the stream has been used to construct `OneOf<FileData, IPartialAttachment>`, the stream has to live until the message and the attachment are sent.
2. That means you either return both an array of `OneOf<FileData, IPartialAttachment>` and the copied `Stream`, or just the copied `Stream`, but in the latter's case, the method will only consist of 4 lines, which makes it pretty useless.
3. I'm not sure if you can `await using` a `Task<OneOf<FileData, IPartialAttachment>[], Stream>`. If not that means we have to manually call `Dispose()` or `DisposeAsync()` in the end of the method, which can be forgot quite easily.

What do you think?

Reply:
Hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

Reply:
This is quite the weakness of C# I've seen 😅

Reply:
I suppose keep it as is for now